### ISSUE TITLE
Replace verbose context field with function

### ISSUE THREAD

**Issue by pjones** [ameingast/postgresql-simple-migration#34]
Would you be open to these two changes:

  1. Use `MonadIO` instead of `IO`, and

  2. Replace the `verbose` context field with a function `(String -> m ())`

These changes would allow integration with existing logging frameworks so migration messages don't go to `stdout` when logs are going somewhere else.

If these changes sound acceptible I'd be happy to supply a PR.

**Comment by ameingast** [comment]
Yes, if it's possible to keep the current public API stable. That means no destructive signature changes to existing public functions or data-types - only extensions.

**Comment by unclechu** [comment]
Implemented #35 in order to solve this.

**Comment by unclechu** [comment]
@ameingast I closed #35 in favor of #36. Please review.
